modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2491,6 +2491,9 @@ public CodegenModel fromModel(String name, Schema schema) {
                 } else { // type is number and without format
                     m.isNumber = Boolean.TRUE;
                 }
+            } else if (ModelUtils.isFreeFormObject(openAPI, schema)) {
+                addAdditionPropertiesToCodeGenModel(m, schema);
+                ModelUtils.syncValidationProperties(schema, m);
```

Comment:
@wing328 should we set `m.isMapModel = true;` here also?

Reply:
Good question. free-form object is a [sub-type of dictionary](https://swagger.io/docs/specification/data-models/dictionaries/) so it may make sense to set `isMapModel` to true but my only concern is how it will impact other generators (e.g. aspnetcore, csharp-netcore, etc)

I would recommend setting it in another PR to see how it impacts other generators.